R/addins.R

```diff
@@ -1,8 +1,123 @@
-read_addin <- function(...) {
-  path <- fs::path_package("rhino", "rstudio", "addins", ...)
-  readChar(path, file.info(path)$size)
+run_background <- function(file) {
+  path <- fs::path_package("rhino", "rstudio", "addins", file)
+  rstudioapi::jobRunScript(path, workingDir = rstudioapi::getActiveProject())
 }
 
 addin_module <- function() {
-  rstudioapi::insertText(read_addin("module.R"))
+  module_path <- fs::path_package("rhino", "rstudio", "addins", "module.R")
+  overwrite <- FALSE
+  file_path <- rstudioapi::selectFile(
+    caption = "Module name and location",
+    path = fs::path("app", "view"),
+    filter = ".R",
+    label = "Save",
+    existing = FALSE
+  )
+  if (is.null(file_path)) {
+    return()
+  }
+  if (tools::file_ext(file_path) == "") {
+    file_path <- glue::glue("{file_path}.R")
+  } else if (fs::file_exists(file_path)) {
+    overwrite <- rstudioapi::showQuestion(
+      title = "File already exists",
+      message = "Would you like to overwrite it?",
+      ok = "Yes",
+      cancel = "No"
+    )
+  }
+  fs::file_copy(module_path, file_path, overwrite = overwrite)
```

Comment:
This part of the code does not work as it should.
1. `selectFile` already asks a user if they want to overwrite the file if it exists. There should be only one such question.
2. Your question fails in case the user does not want to overwrite the file - it shows the error you posted in another place. But it should simply cancel the run (as you already did with `NULL` file path).
3. You miss the case when a user provides a file without an extension - in that case, the addin will not ask for the overwriting. This is also a problem with `selectFile` - it is not going to ask if you provide the name of an existing file but without an extension.